lib/ecto/changeset.ex

```diff
@@ -1395,6 +1395,26 @@ defmodule Ecto.Changeset do
   end
 
   ## Constraints
+  @doc """
+  Checks for a check constraint in the given field.
+
+  The check constraint works by relying on the database to check
+  if the check constraint has been violated or not and, if so,
+  Ecto converts it into a changeset error.
+
+  ## Options
+
+    * `:message` - the message in case the constraint check fails.
+      Defaults to something like "violates check 'products_price_check'"
+    * `:name` - the name of the constraint. Required.
+
+  """
+  def check_constraint(changeset, field, opts \\ []) do
+    Keyword.get(opts, :name) || raise ArgumentError, "must supply a name for the check constraint"
+    constraint = opts[:name]
```

Comment:
Even better, you can have it here:

    constraint = opts[:name] || raise ArgumentError, ...